### PR TITLE
refactor to clear up ServerReactiveSocket logic

### PR DESCRIPTION
**Pull request by yschimke**
not a fix, just making it clearer what is returned

### CODE REVIEW

File: reactivesocket-core/src/main/java/io/reactivesocket/ServerReactiveSocket.java [hunk 1]
```diff
@@ -150,83 +150,88 @@ public ServerReactiveSocket start() {
...
-                        default:
-                            return handleError(streamId, new IllegalStateException("ServerReactiveSocket: Unexpected frame type: "
-                                    + frame.getType()));
-                    }
+                    return handleFrame(frame, streamId);
                 } finally {
                     frame.release();
                 }
             })
-            .doOnError(t -> {
```
Comment (yschimke) [inline reactivesocket-core/src/main/java/io/reactivesocket/ServerReactiveSocket.java]: I'll add this block back

**Comment by robertroeser** [comment]
I'm in the middle of cleaning up the server/client setup and will be sending a pull request for this. Can we wait to make more changes like this until after that?

**Comment by yschimke** [comment]
I'll hold off on this for now

**Comment by yschimke** [comment]
covered by https://github.com/rsocket/rsocket-java/pull/285

**Pull request closed without merging**